src/Formatter/Formatter.php

```diff
@@ -197,9 +197,6 @@ public function getJs()
         $configurator->enableJavaScript();
         $configurator->javascript->exportMethods = ['preview'];
 
-        return $configurator->finalize([
-            'returnParser' => false,
-            'returnRenderer' => false
-        ])['js'];
+        return $configurator->finalize()['js'];
```

Comment:
should `getConfigurator()` result get cached in memory, e.g:

```php
protected function getConfigurator()
{
    if (! $this->configurator) {
        $configurator = new Configurator;
        ...
        ...
        ...
        $this->configurator = $configurator;
    }

    return $this->configurator;
}
```

Reply:
We should probably just be caching `$this->getConfigurator()->finalize()` [here](https://github.com/sijad/core/blob/90dd1bb8b1f9d7a539941b9045cb5151e0dab6e7/src/Formatter/Formatter.php#L153) and refactoring to get the parser/renderer/JS from that.

Reply:
like this?

```php
return $this->cache->rememberForever('flarum.formatter', function () {
    return $this->getConfigurator()->finalize();
});
```

Reply:
Exactly.